sound/soc/sof/core.c

```diff
@@ -400,7 +400,13 @@ int snd_sof_device_shutdown(struct device *dev)
 {
 	struct snd_sof_dev *sdev = dev_get_drvdata(dev);
 
-	return snd_sof_shutdown(sdev);
+	if (IS_ENABLED(CONFIG_SND_SOC_SOF_PROBE_WORK_QUEUE))
+		cancel_work_sync(&sdev->probe_work);
```

Comment:
@plbossart Maybe it is better to move this check into sof_suspend() as sof_suspend() still need this check?